### PR TITLE
Adding tests to work with local file system and updating to latest SDK version.

### PR DESCRIPTION
**Pull request by safihamid**
Updating extensions to use the latest version of ApiHub SDK

Conflicts:
	src/ExtensionsSample/Program.cs
	test/WebJobs.Extensions.Tests/Extensions/ApiHub/ApiHubBindingEndToEndTests.cs

Updates to support PollInterval

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@safihamid__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (hamids). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: src/ExtensionsSample/Program.cs [hunk 1]
```diff
@@ -66,7 +69,8 @@ public static void Main(string[] args)
                 typeof(SendGridSamples),
                 typeof(TableSamples),
                 typeof(TimerSamples),
-                typeof(WebHookSamples));
+                typeof(WebHookSamples),
+                typeof(ApiHubSamples));
```
Comment (mathewc) [inline src/ExtensionsSample/Program.cs]: I'd rather you left this the way it was - when I merged your stuff last week, I moved the ApiHub samples to the same model that the EasyTables and other samples use. See both the sample files - they have doc on how to enable them. The idea is that if/when people want to run them, they can just add this line. This makes the samples much easier to run for people - they don't have to configure the world to get them running.

File: src/WebJobs.Extensions.ApiHub/ApiHubFileTriggerAttribute.cs [hunk 2]
```diff
@@ -9,13 +9,15 @@ namespace Microsoft.Azure.WebJobs
     public sealed class ApiHubFileTriggerAttribute : ApiHubFileAttribute
     {
         /// <summary>
-        ///  Create an instance of an attribute that points to the file on SAAS file trigger
+        /// Create an instance of an attribute that points to the file on SAAS file trigger
```
Comment (mathewc) [inline src/WebJobs.Extensions.ApiHub/ApiHubFileTriggerAttribute.cs]: This attribute description is hard to understand. How about "Attribute used to bind a parameter to a SAAS file"?

File: src/ExtensionsSample/Program.cs (hunk 1 above)
Comment (safihamid) [inline src/ExtensionsSample/Program.cs]: There is no need to configure anything anymore to have ApiHub stuff work, the samples are now much easier and work with local file system. they can be run without ANY extra configuration than other regular stuff.

File: src/WebJobs.Extensions.ApiHub/ApiHubFileTriggerAttribute.cs [hunk 3]
```diff
@@ -9,13 +9,15 @@ namespace Microsoft.Azure.WebJobs
...
         /// </summary>
         /// <param name="key">App settings key name that have the connections string</param>
-        /// <param name="path">Relative path to the file <example>/folder/subfolder/file.txt</example> </param>
-        public ApiHubFileTriggerAttribute(string key, string path)
+        /// <param name="path">Relative path to the file <example>/folder/subfolder/file.txt</example></param>
+        /// <param name="pollIntervalInSeconds">The poll interval in seconds.</param>
+        public ApiHubFileTriggerAttribute(string key, string path, int pollIntervalInSeconds = 0)
```
Comment (mathewc) [inline src/WebJobs.Extensions.ApiHub/ApiHubFileTriggerAttribute.cs]: What does the default value of zero mean? Shouldn't this be a higher value (e.g. 15 seconds?)

File: src/ExtensionsSample/Program.cs (hunk 1 above)
Comment (safihamid) [inline src/ExtensionsSample/Program.cs]: Take a look at CheckForApiHub function below:

 else
 +            {
 +                apiHubConnectionString = "UseLocalFileSystem=true;Path=" + Path.GetTempPath() + "ApiHubDropBox";
 +            }

File: src/ExtensionsSample/Program.cs [hunk 4]
```diff
@@ -112,5 +116,39 @@ private static void EnsureSampleDirectoriesExist(string rootFilesPath)
 
             File.WriteAllText(Path.Combine(rootFilesPath, "input.txt"), "WebJobs SDK Extensions!");
         }
+
+        private static bool CheckForApiHub(JobHostConfiguration config)
```
Comment (mathewc) [inline src/ExtensionsSample/Program.cs]: Recommend removing this - when people enable the sample, they will have configured a valid connection string, so this shouldn't be needed, right?

File: src/WebJobs.Extensions.ApiHub/ApiHubFileTriggerAttribute.cs (hunk 3 above)
Comment (safihamid) [inline src/WebJobs.Extensions.ApiHub/ApiHubFileTriggerAttribute.cs]: 0 means use the default value which is 90 seconds now. look at this code:
        private const int DefaultPollIntervalInSeconds = 90;

            _poll = _folderSource.CreateNewFileWatcher(OnNewFile, _pollIntervalInSeconds > 0 ? _pollIntervalInSeconds : DefaultPollIntervalInSeconds);

File: src/ExtensionsSample/Program.cs (hunk 4 above)
Comment (safihamid) [inline src/ExtensionsSample/Program.cs]: No this means if there is no valid connection string just use the local file system. and then everything just works fine.

File: src/ExtensionsSample/Program.cs (hunk 1 above)
Comment (mathewc) [inline src/ExtensionsSample/Program.cs]: I'd still prefer if you left these opt in as they are now - that's not a problem, right? That's the pattern we're using for all the other extensions we've been onboarding lately: EasyTables, DocumentDB, NotificaitonHubs, etc.

File: src/WebJobs.Extensions.ApiHub/ApiHubFileTriggerAttribute.cs (hunk 3 above)
Comment (mathewc) [inline src/WebJobs.Extensions.ApiHub/ApiHubFileTriggerAttribute.cs]: I see, thanks.

**Comment by mathewc** [comment]
:ship:

**Pull request closed without merging**